### DISCUSSION TITLE
How to add extra payloads in /chat-messages response?

### DISCUSSION THREAD

**Discussion by gaoxiao** [langgenius/dify#4884] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I want to build API services using Studio. Besides providing a direct answer to the user, I also want to add some extra information in the response. Ideally, I want to return something like:

```json
{
  "response": "hello from bot",
  "is_end": true
}
```

However, I can't simply put this format in the Answer node, because it will be fed back to the LLM as history, and I don't want JSON in the history, only the "response" field. Is there a way to include these extra fields in the API response without them appearing in the history?

### 2. Describe the feature you'd like to see

Include extra fields in the API response without them appearing in the history.

### 3. How will this feature improve your workflow or experience?

Include extra fields in the API response without them appearing in the history.

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by patryk20120** [comment]
If you ask if there is a simple way, there is not if it comes to chats. It is meant to work as a chat, for other fields you can use workflow. Maybe combination of those two will help ya?

What is your use case?

**Reply by gaoxiao** [reply]
I want to check the response of LLM to extract some fields if present, and then return the extra fields in response.

**Reply by gaoxiao** [reply]
Seems workflow is a good fit for output format. 
But it doesn't maintain chat history, right?

**Reply by patryk20120** [reply]
No it does not, but you can use workflow as a tool in a chat (e.g. chatflow - which saves conversation history)

**Reply by gaoxiao** [reply]
Sorry I don't get it.
How can I return json nested format while save conversation history? 
I tried adding a workflow node between "Start" and "Answer" node in CHATBOT, but that doesn't seem to work as expected.

**Reply by patryk20120** [reply]
So you want to save this JSON to conversation history?

**Reply by gaoxiao** [reply]
I want to return this json in api call:
{
  "response": "hello from bot",
  "is_end": true
}

but keep only "hello from bot" in conversation history.

**Reply by patryk20120** [reply]
So for as far as I know it is impossible right now. You can return json with workflow, but there is no conversation history. You can use chatbot but you cannot return custom json object.

I think you need a custom solution. What is your use case? Is it necessary?
